.clang-tidy

```diff
@@ -110,3 +112,18 @@ HeaderFilterRegex: "src/esp/"
 #   really improve readability in any way. Keeping this as a warning tho, as it
 #   *might* actually be useful in certain cases, just not treating it as an
 #   error.
+# performance-no-int-to-ptr
+#   "error: integer to pointer cast pessimizes optimization opportunities"
+#   triggered by gfx::replay::NodeHandle, which can be either a pointer (for
+#   the SceneGraph backend) or an arbitrary integer value (for the new
+#   data-oriented backend). Clang Tidy, tell me, how else am I supposed to
+#   implement type-safe handles?!
```

Comment:
With `//NOLINTNEXTLINE`? I'd rather keep this check if possible as it's otherwise useful (mostly for catching types that should ptr_diff.

Reply:
I thought about this, and I don't see a scenario where *explicitly reinterpreting* pointer as an integer would be harming any optimization opportunities. In other words, given that we ban C-style casts with `-Wold-style-cast` anyway, this is hard to do by accident, and if there's a `reinterpret_cast` from an integer to a pointer or back, it's probably intentional -- i.e., to implement a handle type.